mod_reforged/hooks/skills/effects/bleeding_effect.nut

```diff
@@ -0,0 +1,131 @@
+			type = "text",
+			icon = "ui/icons/special.png",
+			text = ::Reforged.Mod.Tooltips.parseString("If at least " + ::MSU.Text.colorRed(this.m.StacksForMoraleCheck) + " stacks are received in a single [turn|Concept.Turn], a negative [morale check|Concept.Morale] is immediately triggered")
+		});
+		return ret;
+	}
+
+	q.getName <- function()
+	{
+		return this.m.Stacks == 1 ? this.m.Name : this.m.Name + " (x" + this.m.Stacks + ")";
+	}
+
+	q.getDescription = @() function()
+	{
+		return this.m.Description;
+	}
+
+	q.getMalusMult <- function()
+	{
+		return 1.0 - ::Math.minf(this.m.MaxMalusMult, this.m.Stacks * this.m.MalusMultPerStack);
+	}
+
+	q.getDamage <- function()
+	{
+		local ret = this.getContainer().hasSkill("trait.bleeder") ? this.m.Stacks * 2 : 1;
+		return this.getContainer().hasSkill("effects.hyena_potion") ? ::Math.floor(ret * 0.5) : ret;
+	}
+
+	// The vanilla file has an empty onUpdate function
+	q.onUpdate = @() function( _properties )
+	{
+		local mult = this.getMalusMult();
+		_properties.InitiativeMult *= mult;
+		_properties.BraveryMult *= mult;
+	}
+
+	// Based on the vanilla function but use this.getDamage() to calculate hitInfo.DamageRegular and use this.m.Overlay for spawnIcon
+	// and get rid of the part that decrements this.m.TurnsLeft and removes the effect
+	// also apply damage in instances smaller than ::Const.Morale.OnHitMinDamage to prevent morale check from this damage
+	q.applyDamage = @() function()
+	{
+		if (this.m.LastRoundApplied != ::Time.getRound())
+		{
+			this.m.LastRoundApplied = ::Time.getRound();
+			local actor = this.getContainer().getActor();
+			this.spawnIcon(this.m.Overlay, actor.getTile());
+
+			// Split the damage instances if damage is greater than ::Const.Morale.OnHitMinDamage
+			// so that the damage does not trigger morale checks
+			local damage = [this.getDamage()];
+			local i = 0;
+			while (i < damage.len() && damage[i] > ::Const.Morale.OnHitMinDamage)
+			{
+				local carryOver = damage[i] - ::Const.Morale.OnHitMinDamage + 1;
+				damage[i] = ::Const.Morale.OnHitMinDamage - 1;
+				damage.push(carryOver);
+				i++;
+			}
+
+			local hitInfo = clone ::Const.Tactical.HitInfo;
+			hitInfo.DamageDirect = 1.0;
+			hitInfo.BodyPart = ::Const.BodyPart.Body;
+			hitInfo.BodyDamageMult = 1.0;
+			hitInfo.FatalityChanceMult = 0.0;
+			foreach (d in damage)
+			{
+				hitInfo.DamageRegular = this.getDamage()
+				actor.onDamageReceived(actor, this, hitInfo);
+			}
+		}
+	}
+
+	q.onRefresh <- function()
+	{
+		local actor = this.getContainer().getActor();
+		if (actor.getCurrentProperties().IsResistantToAnyStatuses && ::Math.rand(1, 100) <= 50)
+		{
+			if (!actor.isHiddenToPlayer())
+				::Tactical.EventLog.log(::Const.UI.getColorizedEntityName(actor) + " had his bleeding wound quickly close thanks to his unnatural physiology");
+			return;
+		}
+
+		this.m.Stacks++;
+		this.spawnIcon(this.m.Overlay, actor.getTile());
+
+		if (++this.m.StacksThisTurn == this.m.StacksForMoraleCheck)
```

Comment:
You must compare using `>=` here

Reply:
We use `==` because we only want to trigger one morale check if `at least 3 bleed stacks were received in a single turn`. `>=` will cause morale checks to trigger on every bleed received starting at 3 and more, that's not intended.

Reply:
Ah, so bleeds are **always** only afflicted one at a time?

Reply:
I'm not sure I fully understand your question. You can only add one instance of bleed at a time but multiple instances may be added in a single attack due to different perks/skills.

Reply:
Ok, I understand. Then the line makes sense to me